app/routes/__index/domains/index.tsx

```diff
@@ -16,13 +22,51 @@ export const loader = async ({ request }: LoaderArgs) => {
 };
 
 export const action = async ({ request }: ActionArgs) => {
-  if (request.method === 'PUT') {
-    // TODO: Call the appropriate code to prolong record
-  } else if (request.method === 'DELETE') {
-    // TODO: Call the appropriate code to delete record
+  const username = await requireUsername(request);
+  const formData = await request.formData();
+  const intent = formData.get('intent');
```

Comment:
Use zod to do the form parsing, like we do in https://github.com/DevelopingSpace/starchart/blob/3fa7f7079b2df10aad33ef3d96450c3fe9dc9c37/app/routes/__index/domains/new.tsx